### DISCUSSION TITLE
Fcal with user devices

### DISCUSSION THREAD

**Discussion by LeidyMoraV** [PlusToolkit/PlusLib#1169] [Q&A] [answered]
Hello, 
I currently have the Ascension Trackstar, Siemens P300 ultrasound, and the Epiphan frame grabber VGA2USB, and I linked all three hardware through PLUS to Slicer. Now I'm working in the calibration, in order to do that. Do I need to write a separate fcal config file? How can I write such configuration?
The config file that I write for the Device Set is the next one:
`<PlusConfiguration version="2.3">

  <DataCollection StartupDelaySec="1.0">
    <DeviceSet 
      Name="PlusServer: Ephiphan and Ascension3D (Reference, Probe, Stylus)"
      Description="Prueba Ephiphan and Ascension 3D" />

    <Device
      Id="TrackerDevice"
      Type="Ascension3DG"
      FilterAcWideNotch="1"
      ToolReferenceFrame="Tracker"
      BaudRate="115200">
      <DataSources>
        <DataSource Type="Tool" Id="Reference" BufferSize="500" PortName="0" AveragedItemsForFiltering="20" />
        <DataSource Type="Tool" Id="Probe" BufferSize="500" PortName="1" AveragedItemsForFiltering="20" />
        <DataSource Type="Tool" Id="Stylus" BufferSize="500" PortName="2" AveragedItemsForFiltering="20" />
      </DataSources>
      <OutputChannels>
        <OutputChannel Id="TrackerStream">
          <DataSource Id="Reference" />
          <DataSource Id="Probe" />
          <DataSource Id="Stylus" />
        </OutputChannel>
      </OutputChannels>
    </Device>
    <Device
      Id="VideoDevice" 
      Type="Epiphan" >
      <DataSources>
        <DataSource Type="Video" Id="Video" PortUsImageOrientation="MF"  ImageType="RGB_COLOR"  />
      </DataSources>      
      <OutputChannels>
        <OutputChannel Id="VideoStream" VideoDataSourceId="Video" />
      </OutputChannels>
    </Device>
    <Device 
      Id="TrackedVideoDevice" 
      Type="VirtualMixer" >
      <InputChannels>
        <InputChannel Id="TrackerStream" />
        <InputChannel Id="VideoStream" />
      </InputChannels>
      <OutputChannels>
        <OutputChannel Id="TrackedVideoStream"/>
      </OutputChannels>
    </Device>   
    <Device
      Id="CaptureDevice"
      Type="VirtualCapture"
      BaseFilename="RecordingTest.igs.mha"
      EnableCapturingOnStart="FALSE" >
      <InputChannels>
        <InputChannel Id="TrackedVideoStream" />
      </InputChannels>
    </Device>
  </DataCollection>

  <CoordinateDefinitions>
    <Transform From="Image" To="Reference"
      Matrix="
        0.2 0.0 0.0 0.0
        0.0 0.2 0.0 0.0
        0.0 0.0 0.2 0.0        
        0 0 0 1" />
  </CoordinateDefinitions>
  
  <PlusOpenIGTLinkServer 
    MaxNumberOfIgtlMessagesToSend="1" 
    MaxTimeSpentWithProcessingMs="50" 
    ListeningPort="18944" 
    SendValidTransformsOnly="true" 
    OutputChannelId="TrackedVideoStream" > 
    <DefaultClientInfo> 
      <MessageTypes> 
        <Message Type="IMAGE" />
        <Message Type="TRANSFORM" />
      </MessageTypes>
      <ImageNames>
        <Image Name="Image" EmbeddedTransformToFrame="Reference" />
      </ImageNames>
      <TransformNames>
        <Transform Name="ProbeToTracker" />
        <Transform Name="ReferenceToTracker" />
        <Transform Name="StylusToTracker" />
      </TransformNames>
    </DefaultClientInfo>
  </PlusOpenIGTLinkServer>

</PlusConfiguration>
`

**Comment by adamrankin** [comment, marked as answer]
Did you want to do calibration in fCal or in Slicer? If you wanted to do a calibration in fCal, I suggest you copy an existing fCal configuration file and modify it with your devices.
